rust/template/differential_datalog/src/program/mod.rs

```diff
@@ -85,7 +89,111 @@ type TKeyAgent<S> = TraceAgent<KeyTrace<S>>;
 type TValEnter<P, T> = TraceEnter<TValAgent<P>, T>;
 type TKeyEnter<P, T> = TraceEnter<TKeyAgent<P>, T>;
 
-/// Diff associated with records in differential dataflow
+#[derive(
+    Abomonation, Copy, Ord, PartialOrd, Eq, PartialEq, Hash, Debug, Clone, Serialize, Deserialize,
+)]
+pub struct CheckedWeight {
```

Comment:
```suggestion
#[derive(
    Abomonation, Copy, Ord, PartialOrd, Eq, PartialEq, Hash, Debug, Clone, Serialize, Deserialize,
)]
#[repr(transparent)]
pub struct CheckedWeight {
```

Reply:
> 
> This PR should configure a subset of the test suite to run with this feature enabled.

Any suggestions? Do we have any tools to measure the perf difference?

Reply:
This is just for correctness, not performance.  Speaking of which, a test that actually overflows and crashes would be nice too.

As far as performance goes, the benchmarks in `differential-datalog/rust/ddlog_benches` is the best we have right now.  (see my other comment).

Reply:
The test overflow.dl crashes with this feature. It is based on the bug report in #878